pkg/dbal/DbalConsumer.php

```diff
@@ -115,33 +97,54 @@ public function reject(Message $message, bool $requeue = false): void
 
             return;
         }
+
+        $this->deleteMessageByDeliveryId($message->getDeliveryId());
```

Comment:
rename to deleteMessage. the method should accept a message instance and throws an exception if delivery id is not set.